File: app/app/src/main/java/me/williamhester/reddit/inject/ActivityComponent.java
package me.williamhester.reddit.inject;

import dagger.Component;
import me.williamhester.reddit.ui.activities.BaseActivity;
import me.williamhester.reddit.ui.fragments.BaseFragment;

/** An Activity-based scope component, to be used once for each Activity lifecycle */
@Component(
    dependencies = ApplicationComponent.class,
    modules = { ApiModule.class, ActivityModule.class })
@ActivityScope
public interface ActivityComponent {
  void inject(BaseFragment fragment);
  void inject(BaseActivity activity);
}

File: app/app/src/main/java/me/williamhester/reddit/inject/ActivityComponent.kt
package me.williamhester.reddit.inject

import dagger.Subcomponent
import me.williamhester.reddit.ui.activities.BaseActivity
import me.williamhester.reddit.ui.fragments.BaseFragment

/**
 * A Subcomponent of [ApplicationComponent] that should be instantiated each time an Activity is
 * started for the first time.
 */
@Subcomponent(modules = arrayOf(ActivityModule::class))
@ActivityScope
interface ActivityComponent {
  fun inject(fragment: BaseFragment)
  fun inject(activity: BaseActivity)
}
